src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -106,13 +122,23 @@ public override BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapInd
             {
                 if (mapIndices.DataType != typeof(long))
                     throw new ArgumentException(Strings.MismatchedValueType + " PrimitiveColumn<long>", nameof(mapIndices));
-                if (mapIndices.Length != Length)
+                if (mapIndices.Length >= Length)
                     throw new ArgumentException(Strings.MismatchedColumnLengths, nameof(mapIndices));
                 return Clone(mapIndices as PrimitiveColumn<long>, invertMapIndices);
             }
             return Clone();
         }
 
+        internal override BaseColumn CloneAndAppendNulls(BaseColumn mapIndices = null, bool invertMapIndices = false)
+        {
+            PrimitiveColumn<T> ret = Clone(mapIndices, invertMapIndices) as PrimitiveColumn<T>;
+            for (long i = 0; i < NullCount; i++)
+            {
+                ret.Append(null);
```

Comment:
Instead of appending `null` multiple times, what do you think about a method that will do it all at once?  Like `AppendMany(value, count)`.

Reply:
Yup, good idea. Done